go/worker/keymanager/churp.go

```diff
@@ -76,12 +104,47 @@ func (w *churpWorker) Methods() []string {
 }
 
 // Connect implements RPCAccessController interface.
-func (w *churpWorker) Connect(context.Context, core.PeerID) bool {
+func (w *churpWorker) Connect(_ context.Context, peerID core.PeerID) (b bool) {
+	// Secure methods are accessible to peers that pass authorization.
+	if err := w.authorizeKeyManager(peerID); err == nil {
+		return true
+	}
+
 	return false
 }
 
 // Authorize implements RPCAccessController interface.
-func (w *churpWorker) Authorize(context.Context, string, enclaverpc.Kind, core.PeerID) error {
+func (w *churpWorker) Authorize(_ context.Context, method string, kind enclaverpc.Kind, peerID core.PeerID) (err error) {
+	// Check if the method is supported.
+	switch kind {
+	case enclaverpc.KindInsecureQuery:
+		if _, ok := insecureChurpRPCMethods[method]; !ok {
+			return fmt.Errorf("unsupported method: %s", method)
+		}
+		return nil
```

Comment:
Matrix is of size `(t+1) x (2t+1)`. For `t=4` (you need at least 5 key manager nodes to reconstruct the secret), that is 45 group elements that need to be converted to bytes and appended to an array. In p384, that should be `~45*384 bits = ~2KB`.

Since only new committee nodes need this, I removed this line (`return nil`) so all nodes need to be authorized.